### ISSUE TITLE
Document content of the various "Listings Items" calls

### ISSUE THREAD

**Issue by johnkw** [amzn/selling-partner-api-docs#1025]
The documentation on https://github.com/amzn/selling-partner-api-docs/blob/main/references/listings-items-api/listingsItems_2020-09-01.md is missing for almost every parameter.

I was hoping to find an SP-API replacement for using the "Add Products via Upload" offers feed.  At this point the documentation is so vague and unclear, that I'm not even certain if "Listings Items" calls will even do that.  Do they?

As for specific documentation missing:

* `productType` says "The Amazon product type of the listings item.", but doesn't say what that is.  What are the options for that string?

* What is `path`? How can we get the list of valid paths?

* `value` is a `< object > array`, which doesn't explain much.  What is an "object"?  I would guess it's an arbitrary dictionary of JSON key/value pairs, but... if so, what are the valid keys and values?

**Comment by chapmanjw** [comment]
The use case guide for the Listings Item APIs provides more details on how the APIs can be used: https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/listings-items-api-use-case-guide/listings-items-api-use-case-guide_2020-09-01.md.

The product types and related data requirements (which define the attributes data handled by the Listings Item APIs) are available in the Product Type Definitions APIs: https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/product-type-definitions-api-use-case-guide/definitions-product-types-api-use-case-guide_2020-09-01.md.

**Comment by SouravPal95** [comment]
@chapmanjw 
I have been trying to use the feeds API to create product listings. However, any `productType` value I provided rendered a processing report with an error of invalid or unsupported product type. So I decided to use the search product definitions API endpoint for marketplace ID (A21TJRUUN4KGV) for India. It returned empty for LUGGAGE and every other keyword that I tried. Then I tried without any keyword and received PRODUCT as the only available type. Then on using it with `requirements` attribute `LISTING`, I received a processing report with error message that cannot use product type PRODUCT to create products which is an a problem because that is all I have available..

**Comment by johnkw** [comment]
Interesting thanks! Those other documents are extremely well hidden, so good job finding them @chapmanjw somehow. I do see the magic `path` is apparently something like `/attributes/purchasable_offer`, although I didn't immediately find how that's constructed.  It doesn't seem to be documented anywhere, except in being included in that example.

It would be good to merge listings-items-api-use-case-guide_2020-09-01.md directly in to the main API document, since it has the only explanation of what any of the calls do.

**Comment by chapmanjw** [comment]
@SouravPal95 Yes, we have migrated some product types for sellers (merchants) in the US and EU5 marketplaces. We have more broad coverage for vendors in other marketplaces, but have not yet migrated content for sellers in the IN marketplace. For marketplaces without migrated product types, we are currently limited to editing common attributes, such as price and inventory, but do not yet support creating new products without the migrated product types.

@johnkw That is good feedback. I will work with our technical writers on how we can improve that (if nothing else, linking the two to each other).

**Comment by SouravPal95** [comment]
@chapmanjw I really appreciate your reply. This will actually allow us to move forward in the right direction.

**Comment by TKwheat** [comment]
Hello, now you have successfully uploaded the listing through this api? I got the following response:
"message": "Access to requested resource is denied."

**Comment by chapmanjw** [comment]
@TKwheat What API request are you making that results in the access denied error?

**Comment by ekindev** [comment]
@chapmanjw Did you migrate product types for Turkey? I have same issue.

**Comment by chapmanjw** [comment]
@ekindev As of today, only the `PRODUCT` product type is available in the TR marketplace for merchants (which allows for editing offer details and some common product attributes). ~120 product types have been migrated in US and EU5 marketplaces. I don't have any specific timelines for migrating content in the TR marketplace.

**Comment by ekindev** [comment]
@chapmanjw Why am I still getting this error while trying to create product: "Creating products is not supported with the \"PRODUCT\" Amazon product type. Either specify a specific Amazon product type or specify an offer-only requirements set."

**Comment by ekindev** [comment]
@chapmanjw Hello John, we are still waiting for you. Is MWS API still open for the new applications?

**Comment by ekindev** [comment]
@chapmanjw Hello John, we are still waiting for you. Is MWS API can be used for the new applications?

**Comment by chapmanjw** [comment]
> @chapmanjw Hello John, we are still waiting for you. Is MWS API can be used for the new applications?

What specifically are you asking about on the MWS APIs? For creating new products, until content is migrated for the TR marketplace, the Category XSDs can be used to submit XML feeds with the SP Feeds API.

**Comment by ekindev** [comment]
> > @chapmanjw Hello John, we are still waiting for you. Is MWS API can be used for the new applications?
> 
> What specifically are you asking about on the MWS APIs? For creating new products, until content is migrated for the TR marketplace, the Category XSDs can be used to submit XML feeds with the SP Feeds API.

I was think that MWS APIs can be used as alternative to SP-API but seems not likely possible for the new seller applications. 

Anyway, thanks for the answer.

**Comment by sojujs** [comment]
Hi @chapmanjw , regarding the issue raised by @SouravPal95 are product types in Indian marketplaces supported now to upload if not then when can we expect it ? And till then is it possible to upload different types of products using the feeds API ? 

Since if i try to search product type definitions i can only find "PRODUCT" for the Indian marketplace.

Also if we have no options whatsoever from SP-API to create product listings, how can we do it programmatically? AFAIK newer sellers cannot signup with MWS APIs anymore.

**Comment by johnkw** [comment]
The "feeds" system is still the way to update SKUs with every parameter. It's certainly buggy though. And when anything fails to update, Seller Support is unable to give answers.

**Comment by shirushi-dev** [comment]
I hope this feature will be improved soon

**Comment by johnkw** [comment]
I just noticed there's a new API version document. Although the documentation is still missing for `productType` in https://github.com/amzn/selling-partner-api-docs/blob/main/references/listings-items-api/listingsItems_2021-08-01.md#patchoperation , at least you can call `getListingsItem` for a current SKU to see what it should be.

**Comment by shirushi-dev** [comment]
@johnkw 
Is it impossible to list new products for "patch operation"?

**Comment by shirushi-dev** [comment]
"product_type" => "PRODUCT", but I get the following

    "issues": [
        {
            "code": "4000004",
            "message": "Creating products is not supported with the \"PRODUCT\" Amazon product type. Either specify a specific Amazon product type or specify an offer-only requirements set.",
            "severity": "ERROR"
        }
    ]


Therefore, the listing function is not available.
I don't understand.

**Comment by shirushi-dev** [comment]
Has this been improved? Please tell us.

**Comment by johnkw** [comment]
The documentation has moved to https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference but is still broken in that it says:
productType  required | The Amazon product type of the listings item. | string

`string` is not helpful. There needs to be a link to some place to get the complete enumeration of all valid values.

**Comment by GJ-JG** [comment]
Hi all, 

Can you use  [Product Type Definitions API v2020](https://developer-docs.amazon.com/sp-api/docs/product-type-definitions-api-v2020-09-01-reference)  to fetch the ```ProductType``` and then programmatically use it in [Listings Items API v2021-08-01](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference)? In this way you can get the latest product type definitions.

Hope this helps.

Best,
Jiang

**Comment by johnkw** [comment]
Yes the documentation in places like these:
https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#listingsitemputrequest
https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#listingsitempatchrequest
needs to link over to the documentation at:
https://developer-docs.amazon.com/sp-api/docs/product-type-definitions-api-v2020-09-01-reference
and the former locations need to explicitly state what those are.

**Comment by verlonsalaysay** [comment]
Hi all,

Is there a way for us to be able to perform a patch operation in the **status** of a Listing, like the one below? e.g. change status from DISCOVERABLE to BUYABLE.

Apparently, I am not able to send a Patch request using the path to point to the status attribute of the listing. Below is an example of what I can get from the GET Listing request.

"sku": "2608690145",
    "summaries": [
        {
            "marketplaceId": "A1PA6795UKMFR9",
            "asin": "B0053PWE60",
            "productType": "CHISEL",
            **"status": [
                "DISCOVERABLE"
            ],**

 This is the Body of my patch request contains that is trying to replace the status of the listing:

  "productType":"CHISEL",
  "patches":[
    {
      "op":"replace",
      "path":"/status",
      "value":[
        {
                "value": "BUYABLE"
        }

**Comment by johnkw** [comment]
@verlonsalaysay please delete that and move it to a new ticket. It has nothing to do with this ticket.

**Comment by johnkw** [comment]
Still broken as per my https://github.com/amzn/selling-partner-api-docs/issues/1025#issuecomment-1508586641 comment.
